WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/pages/PagesCardViewHolder.kt

```diff
@@ -22,11 +25,22 @@ class PagesCardViewHolder(
 
     fun bind(card: PagesCard) = with(binding) {
         val pagesCard = card as PagesCardWithData
-        (pagesItems.adapter as PagesItemsAdapter).update(pagesCard.pages)
         mySiteToolbar.update(pagesCard.title)
+        if (pagesCard.pages.isEmpty()) pagesItems.visibility = View.GONE
```

Comment:
❓ Could the visibility for pagesItems be determined in the builder rather than in the viewholder? wdyt?

Reply:
On revisiting this logic, I found out there is no need to set the visibility. If there are no items in the `PagesItemsAdapter`. the`onBindViewHolder` is not called, and hence no space is taken up by the recycler view. I have tested this logic by setting a red background to the recycler view and page item view and removed the visibility logic. When there are no pages, I can't see any red background. 👍🏼 

So I am not adding any visibility logic here. Still, I have refactored the logic of updating the items in `PagesItemsAdapter` here in this commit [420f0e3](https://github.com/wordpress-mobile/WordPress-Android/pull/18322/commits/420f0e34b0a5fb2851ab82ca27102723cf155e5d). Let me know if this makes sense.